### ISSUE TITLE
pdf:metadata does not accept unicode

### ISSUE THREAD

**Issue by Omikhleia** [sile-typesetter/sile#1280]
SILE 0.12

Unicode characters in metadata...
```
\pdf:metadata[key=Author, val=Erwin Schrödinger]
\pdf:metadata[key=Title, val=Erwin Schrödinger’s cat]
```
... do not work well:

![image](https://user-images.githubusercontent.com/18075640/141337737-8cfa328b-6b27-45ca-ab2e-c222e90a2a05.png)

The PDF standard says:
> The text string type shall be used for character strings that shall be encoded in either PDFDocEncoding or the 
UTF-16BE  Unicode  character  encoding  scheme.  PDFDocEncoding  can  encode  all  of  the  ISO  Latin  1  
character set and is documented in Annex D. UTF-16BE can encode all Unicode characters.

This (very) convoluted way indeed works, confirming that UTF16BE is the solution:

```
\script{
local pdf = require("justenoughlibtexpdf")
SILE.outputters.libtexpdf._init()

local info = pdf.get_dictionary("Info")
pdf.add_dict(info, pdf.parse("/Title"), 
     pdf.parse("<"..SU.utf8_to_utf16be_hexencoded("Erwin Schrödinger’s cat")..">"))
}
```
![image](https://user-images.githubusercontent.com/18075640/141348780-5132659e-198e-4eb6-be5a-d0f9a9e8ca6c.png)

I tried to find an easier workaround, but `pdf_metadata` in the C code does some `strlen()` https://github.com/sile-typesetter/sile/blob/master/src/justenoughlibtexpdf.c#L328 so simply passing the UTF16LE-encoded string there cannot work. It should likely have been a `lua_rawlen(L, 2)` there - but I haven't found yet how to compile this library.

_In other terms_, solving this issue likely implies
- fixing the C code in justenoughlibtexpdf
- encoding in UTF16BE in the Lua "pdf" package

**Comment by Omikhleia** [comment]
> This (very) convoluted way indeed works, confirming that UTF16BE is the solution:

Well, not so convoluted: this is what the C function does more or less, so using that in the pdf package would work without needed to touch the C code. I dunno, depends on how low level we want to be with respect to PDF tags :astronaut: :

**Comment by Omikhleia** [comment]
While I am at it, the package doc says
> Common keys include \code{Title}, \code{Author}, \code{Subject}, \code{Keywords}, \code{CreationDate}, and
\code{ModDate}.

The format for the two latter, however, should be a date conforming to `(D:YYYYMMDDHHmmSSOHH ' mm)` - would be neat to ensure that. All other fields (incl. ones not specified in the standard) are text string, except "/Trapped", which corresponds to something very special (we could reject it, just because it's likely nonsensical from user input).

**Comment by ctrlcctrlv** [comment]
@Omikhleia There were actually multiple problems here, solved in #1281…

① BOM wasn't being added. BOM is required by the standard to recognize a hex-encoded UTF16 string as one and not intentional Latin-1 title.

![image](https://user-images.githubusercontent.com/838783/141440258-e2462626-8398-4363-8c34-de797682687c.png)
![image](https://user-images.githubusercontent.com/838783/141440316-a0a3ea1c-d9af-4d6c-8581-bea5a25880ef.png)

Not sure what viewer you're using, but Okular refused to recognize metadata without BOM as UTF-16.

② As you noted, we weren't attempting to UTF16-encode at all.
